### PR TITLE
Updates sidebar toggle, theme, and layout to rely on Chakra UI

### PR DESCRIPTION
**Pull request by CtrlSpice**
Sidebar-collapsed
Sidebar-expanded
Sidebar-dark

### CODE REVIEW

File: desktop-exporter/app/routes/main-view.tsx [hunk 1]
```diff
@@ -29,57 +36,64 @@ function Row({ index, style, data }: RowProps) {
...
+    sidebarWidth = "250px";
+    buttonIcon = <ArrowLeftIcon />;
+    traceList = (
+      <FixedSizeList
+        className="list"
+        height={500}
+        itemData={traceSummaries}
+        itemCount={traceSummaries.length}
+        itemSize={30}
+        width="100%"
```
Comment (jmorrell) [inline desktop-exporter/app/routes/main-view.tsx:61]: I believe the list may be being squeezed as it collapses. We might want to put the list into a fixed width box so the container shrinking doesn't affect its layout

File: desktop-exporter/app/routes/main-view.tsx [hunk 2]
```diff
@@ -29,57 +36,64 @@ function Row({ index, style, data }: RowProps) {
...
+      transition="width 0.2s ease-in-out"
+      width={sidebarWidth}
+    >
+      <Flex justifyContent={"flex-end"}>
+        <IconButton
+          aria-label="Expand Sidebar"
+          colorScheme="pink"
+          icon={buttonIcon}
+          margin="15px"
+          onClick={() => props.toggle()}
```
Comment (jmorrell) [inline desktop-exporter/app/routes/main-view.tsx:81]: I think you could simplify this to:
```suggestion
          onClick={props.toggle}
```

File: desktop-exporter/app/routes/main-view.tsx [hunk 3]
```diff
@@ -29,57 +36,64 @@ function Row({ index, style, data }: RowProps) {
...
-        </FixedSizeList>
-      </nav>
-    </div>
+    <Flex
+      bgColor={sidebarColour}
+      direction="column"
+      transition="width 0.2s ease-in-out"
+      width={sidebarWidth}
+    >
+      <Flex justifyContent={"flex-end"}>
```
Comment (jmorrell) [inline desktop-exporter/app/routes/main-view.tsx:75]: nit: if you are going to use a string, you don't need the curly brackets

```suggestion
      <Flex justifyContent="flex-end">
```

**Review by jmorrell — APPROVED** [review]
DALL·E 2022-11-28 17 58 38 - corgi looking for approval, digital art

File: desktop-exporter/app/routes/main-view.tsx (hunk 1 above)
Comment (CtrlSpice) [reply desktop-exporter/app/routes/main-view.tsx:61]: I will fix this thing in the next PR, when I style the list itself.

File: desktop-exporter/app/routes/main-view.tsx [hunk 4]
```diff
@@ -1,7 +1,14 @@
 import React from "react";
 import { Outlet, NavLink, useLoaderData } from "react-router-dom";
 import { FixedSizeList } from "react-window";
-import { useToggle } from "usehooks-ts";
```
[the remark this answers is not in the document]
Comment (CtrlSpice) [reply desktop-exporter/app/routes/main-view.tsx:4]: So, you know how Go just cleans up after itself when you change your mind about a dependency? I really like that. It's really nice.

File: desktop-exporter/app/routes/main-view.tsx (hunk 3 above)
Comment (CtrlSpice) [reply desktop-exporter/app/routes/main-view.tsx:75]: Autocomplete was trying to be helpful 😭

File: desktop-exporter/app/routes/main-view.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jmorrell) [reply desktop-exporter/app/routes/main-view.tsx:4]: yes, it is very nice and I miss it

**Pull request merged**